libogc/sys_report.c

```diff
@@ -99,19 +104,38 @@ void SYS_STDIO_Report(bool use_stdout)
 	}
 }
 
-
 void SYS_Report (char const *const fmt_, ...)
 {
-
-	size_t len;
-
+	ssize_t len = 0;
+	char* buffer = NULL;
 	va_list args;
+	size_t write_len = __outsz;
+	while(true)
+	{
+		char* tmp = (char*)realloc(buffer, write_len);
+		if(tmp == NULL)
+			break;
+
+		buffer = tmp;
+
+		va_start(args, fmt_);
+		len = vsnprintf(buffer, write_len, fmt_, args);
+		va_end(args);
+		if(len >= write_len)
```

Comment:
also check for negative return & bail out

Reply:
resolved, added len > 0 check because the signed difference could again cause issues here :)